prover/prover_fri_utils/src/socket_utils.rs

```diff
@@ -19,7 +19,7 @@ pub fn send_assembly(
     let mut error_messages = vec![];
 
     for _ in 0..10 {
-        match TcpStream::connect(socket_address) {
+        match TcpStream::connect_timeout(socket_address, Duration::from_secs(6)) {
```

Comment:
Nit: In the PR description you mention that we limit the time to 1 minute here, and I think it'd make a great comment. I first looked at diff and wondered "Why 6 seconds though?", then looked at the PR description, and then it all made sense.